docs/tracing.md

```diff
@@ -85,3 +85,102 @@ func clientWithTrace() error {
     defer span.End()
     ...
 }
+```
+## Manual instrumentation
+
+OpenTelemetry provides language specific [API](https://pkg.go.dev/go.opentelemetry.io/otel) libraries to instrument parts of your application that are not covered by automatic instrumentation.
+
+In Containerd, a thin wrapper library defined in `tracing/tracing.go` provides additional functionality and makes it easier to use the OpenTelemetry API for manual instrumentation.
+
+### Creating a new span
+
+To create a new span, use the `tracing.StartSpan()` method. You should already have a global TracerProvider set by configuring `io.containerd.tracing.processor.v1.otlp` plugin, else this will only create an instance of a NoopSpan{}.
+
+```
+func CreateContainer(ctx context.Context, r *runtime.CreateContainerRequest) error {
+    ctx, span := tracing.StartSpan(ctx,
+        tracing.Name(criSpanPrefix, "CreateContainer") // name of the span
+        tracing.WithAttribute("sandbox.id",r.GetPodSandboxId(), //attributes to be added to the span
+        )
+	defer span.End() // end the span once the function returns
+    ...
+}
+```
+Mark the span complete at the end of workflow by calling `Span.End()`. In the above example, we use 'defer' to ensure that the span is properly closed and its duration is recorded.
+
+### Adding Attributes to a span
+
+You can add additional attributes to the span using `Span.SetAttributes()`. Attributes can be added during span creation (by passing `tracing.WithAttribute()` to tracing.StartSpan()) or at any other time during the lifecycle of a span before it has completed.
+
+```
+func CreateContainer(ctx context.Context, r *runtime.CreateContainerRequest) error {
+    ctx, span := tracing.StartSpan(ctx,
+        tracing.Name(criSpanPrefix, "CreateContainer")
+        tracing.WithAttribute("sandbox.id",r.GetPodSandboxId(),
+        )
+	defer span.End()
+    ...
+    containerId := util.GenerateID()
+    containerName := makeContainerName(metadata, sandboxConfig.GetMetadata())
+
+    //Add new attributes to the existing span
+    span.SetAttributes(
+		tracing.Attribute("container.id", containerId),
+		tracing.Attribute("container.name", containerName),
+	)
+    ...
+}
+```
+### Adding an Event to a span
+Use `Span.AddEvent()` to add an event to an existing span. A [span event](https://opentelemetry.io/docs/instrumentation/go/manual/#events) is a specific occurrence within a span, such as the completion of an operation or the occurrence of an error. Span events can be used to provide additional information about the operation represented by the span, and can be used for debugging or performance analysis.
```

Comment:
please call out `span.RecordError` in this paragraph.

Reply:
Sure I can add. But `span.RecordError` impl still need to be merged https://github.com/containerd/containerd/pull/7448

Reply:
Added a separate section about recording errors in a span.